pkg/tfbridge/provider.go

```diff
@@ -514,7 +514,7 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 	// After all is said and done, we need to go back and return only what got populated as a diff from the origin.
 	pinputs := MakeTerraformOutputs(inputs, res.TF.Schema, res.Schema.Fields, assets, false, p.supportsSecrets)
 	minputs, err := plugin.MarshalProperties(pinputs, plugin.MarshalOptions{
-		Label: fmt.Sprintf("%s.inputs", label), KeepUnknowns: true})
+		Label: fmt.Sprintf("%s.inputs", label), KeepUnknowns: true, KeepSecrets: true})
```

Comment:
`KeepSecrets: p.supportsSecrets`

...ironically, that would have been enough with the first change :)